File: app/src/main/java/com/elegion/android/ui/base/view/ErrorView.java
package com.elegion.android.ui.base.view;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

/**
 * @author Mikhail Barannikov
 */
@StateStrategyType(OneExecutionStateStrategy.class)
public interface ErrorView {

    void showNetworkError();

    void showUnexpectedError();

    void showErrorMessage(@NonNull String message);

    void showErrorMessage(@StringRes int message);

    void hideErrorMessage();
}

File: app/src/main/kotlin/com/elegion/android/template/ui/base/view/ErrorView.kt
package com.elegion.android.template.ui.base.view

import android.support.annotation.StringRes

import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

@StateStrategyType(OneExecutionStateStrategy::class)
interface ErrorView {

    fun showNetworkError()

    fun showUnexpectedError()

    fun showErrorMessage(message: String)

    fun showErrorMessage(@StringRes message: Int)

    fun hideErrorMessage()
}
